### PR TITLE
Add MLDSA87 provisionary templates

### PR DESCRIPTION
**Pull request by ArthurHeymans**

### CODE REVIEW

File: drivers/src/persistent.rs [hunk 1]
```diff
@@ -47,62 +48,87 @@ pub type AuthManifestImageMetadataList =
...
-        self.csr_len
     }
+}
 
-    /// Check if the CSR was unprovisioned
-    pub fn is_unprovisioned(&self) -> bool {
-        self.csr_len == Self::UNPROVISIONED_CSR
-    }
+// [TODO][CAP2] not needed for Mldsa87
+macro_rules! impl_idevid_csr {
```
Comment (swenson) [inline drivers/src/persistent.rs:83]: Maybe use generics instead of macros?

**Review by swenson — DISMISSED** [review]
LGTM

File: drivers/src/persistent.rs (hunk 1 above)
Comment (ArthurHeymans) [reply drivers/src/persistent.rs]: Removed it for mldsa as creating a copy on the stack is expensive.

File: x509/src/fmc_alias_cert_mldsa_87.rs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    fmc_alias_cert.rs
+
+Abstract:
+
+    ECC384 FMC Alias Certificate related code.
```
Comment (mhatrevi) [inline x509/src/fmc_alias_cert_mldsa_87.rs:11]: Comment: MLDSA87

File: x509/src/idevid_csr_mldsa_87.rs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    idevid_csr.rs
```
Comment (mhatrevi) [inline x509/src/idevid_csr_mldsa_87.rs:7]: File name

File: x509/src/ldevid_cert_mldsa_87.rs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    ldevid_cert.rs
```
Comment (mhatrevi) [inline x509/src/ldevid_cert_mldsa_87.rs:7]: Filename

File: x509/src/rt_alias_cert_mldsa_87.rs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    idevid_csr.rs
```
Comment (mhatrevi) [inline x509/src/rt_alias_cert_mldsa_87.rs:7]: Filename

**Review by mhatrevi — APPROVED** [review]

**Pull request merged**